### PR TITLE
Schema for lists (first we would migrate relationships to this)

### PR DESCRIPTION
**Pull request by mrclay**
- relationship.name => list.type
- relationship.guid_two => list.target_guid
- relationship.guid_one => list_items.item_guid

Eventually we'd roll ACLs into this as well.

### CODE REVIEW

File: engine/schema/mysql.sql [hunk 1]
```diff
@@ -2,6 +2,34 @@
...
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `type_id` int(11) NOT NULL,
+  `target_guid` bigint(20) unsigned NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `idx` (`type_id`,`target_guid`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- items in lists (item can be in a single list no more than once)
+CREATE TABLE `prefix_list_items` (
+  `position` int(11) unsigned NOT NULL AUTO_INCREMENT,
```
Comment (ewinslow) [inline engine/schema/mysql.sql]: Position and weight?

**Comment by ewinslow** [comment]
It's beautiful, though do we really want to do the db migration first? What is the actual benefit of db migration vs a higher level php API?

**Comment by mrclay** [comment]
Yeah, ignore this for now.

**Comment by ewinslow** [comment]
I do really like it a lot though.

**Pull request closed without merging**